oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java

```diff
@@ -1302,16 +1307,16 @@ public boolean isBasedOnNtBase() {
             return JcrConstants.NT_BASE.equals(baseNodeType);
         }
 
-        private Map<String, PropertyDefinition> collectPropConfigs(NodeState config,
-                                                                   List<NamePattern> patterns,
-                                                                   List<Aggregate.Include> propAggregate,
-                                                                   List<PropertyDefinition> nonExistentProperties,
-                                                                   List<PropertyDefinition> existentProperties,
-                                                                   List<PropertyDefinition> nodeScopeAnalyzedProps,
-                                                                   List<PropertyDefinition> functionRestrictions,
-                                                                   List<PropertyDefinition> syncProps,
-                                                                   List<PropertyDefinition> similarityProperties) {
-            Map<String, PropertyDefinition> propDefns = new HashMap<>();
+        private TreeMap<String, PropertyDefinition> collectPropConfigs(NodeState config,
+                                                                       List<NamePattern> patterns,
+                                                                       List<Aggregate.Include> propAggregate,
+                                                                       List<PropertyDefinition> nonExistentProperties,
+                                                                       List<PropertyDefinition> existentProperties,
+                                                                       List<PropertyDefinition> nodeScopeAnalyzedProps,
+                                                                       List<PropertyDefinition> functionRestrictions,
+                                                                       List<PropertyDefinition> syncProps,
+                                                                       List<PropertyDefinition> similarityProperties) {
+            TreeMap<String, PropertyDefinition> propDefns = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
I'm not sure if TreeMap with CASE_INSENSITIVE_ORDER is faster than HashMap... I know, toLowerCase is not needed, but then, aren't most property names already lowercase? So toLowerCase() will just return "this".

Reply:
I have noticed this optimization opportunity while profiling a run of the indexing job, where the calls to `toLowerCase()` were responsible for a significant part of object allocation in that run, and also showed up on CPU profiling. Property names are usually Camel case, so most of them will in fact have to be converted to lower case as most consist of two or more logical words.